# Importing Libraries and Loading Data

In [131]:
import pandas as pd
df = pd.read_csv('./data/Aviation_Data.csv')
df

<ipython-input-131-1b1420092910>:2: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('./data/Aviation_Data.csv')


Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country   Latitude  Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1       BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2        Saltville, VA  United States  36.922223 -81.878056          NaN   
3           EUREKA, CA  United States        NaN        NaN          NaN   
4           Canton, OH  United States        NaN        NaN          NaN   
...                ...            ...        ...        ...          ...   
90343    Annapolis, MD  United States        NaN        NaN          NaN   
90344      Hampton, NH  United States        NaN        NaN          NaN   
90345       Payson, AZ  United States    341525N   1112021W          PAN   
90346       Morgan, UT  United States        NaN        NaN          NaN   
90347       Athens, GA  United States        NaN        NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               N

# Checking Data Structure, Shape and Columns

In [132]:
df.shape

(90348, 31)

In [133]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50249 non-null  object 
 9   Airport.Name            52790 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87572 non-null  object 
 14  Make                    88826 non-null

In [134]:
df.describe(include='object')

Event.Id Investigation.Type Accident.Number  Event.Date  \
count            88889              90348           88889       88889   
unique           87951                 71           88863       14782   
top     20001212X19172           Accident      CEN22LA149  1984-06-30   
freq                 3              85015               2          25   

             Location        Country Latitude Longitude Airport.Code  \
count           88837          88663    34382     34373        50249   
unique          27758            219    25592     27156        10375   
top     ANCHORAGE, AK  United States  332739N  0112457W         NONE   
freq              434          82248       19        24         1488   

       Airport.Name  ... Amateur.Built    Engine.Type FAR.Description  \
count         52790  ...         88787          81812           32023   
unique        24871  ...             2             13              31   
top         Private  ...            No  Reciprocating             091   
freq            240  ...         80312          69530           18221   

       Schedule Purpose.of.flight Air.carrier Weather.Condition  \
count     12582             82697       16648             84397   
unique        3                26       13590                 4   
top        NSCH          Personal       Pilot               VMC   
freq       4474             49448         258             77303   

       Broad.phase.of.flight   Report.Status Publication.Date  
count                  61724           82508            73659  
unique                    12           17007             2923  
top                  Landing  Probable Cause       25-09-2020  
freq                   15428           61754            16317  

[4 rows x 26 columns]

# Cleaning Data

In [135]:
# Checking for Duplicated Values
df.duplicated()

0        False
1        False
2        False
3        False
4        False
         ...  
90343    False
90344    False
90345    False
90346    False
90347    False
Length: 90348, dtype: bool

In [136]:
# dropping the duplicates
df.drop_duplicates()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country   Latitude  Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1       BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2        Saltville, VA  United States  36.922223 -81.878056          NaN   
3           EUREKA, CA  United States        NaN        NaN          NaN   
4           Canton, OH  United States        NaN        NaN          NaN   
...                ...            ...        ...        ...          ...   
90343    Annapolis, MD  United States        NaN        NaN          NaN   
90344      Hampton, NH  United States        NaN        NaN          NaN   
90345       Payson, AZ  United States    341525N   1112021W          PAN   
90346       Morgan, UT  United States        NaN        NaN          NaN   
90347       Athens, GA  United States        NaN        NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               N

In [137]:
# Changing date to date type and grouping my data

df['Event.Date'] = pd.to_datetime(df['Event.Date'], errors='coerce')
df['Event.Date'].isna().sum

<bound method NDFrame._add_numeric_operations.<locals>.sum of 0        False
1        False
2        False
3        False
4        False
         ...  
90343    False
90344    False
90345    False
90346    False
90347    False
Name: Event.Date, Length: 90348, dtype: bool>

In [138]:
# Grouping Categorical columns and numerical columns
categorical_cols = df.select_dtypes(include=['object']).columns.tolist()
numerical_cols = df.select_dtypes(include=['int64', 'float64']).columns.tolist()

In [139]:
categorical_cols

['Event.Id',
 'Investigation.Type',
 'Accident.Number',
 'Location',
 'Country',
 'Latitude',
 'Longitude',
 'Airport.Code',
 'Airport.Name',
 'Injury.Severity',
 'Aircraft.damage',
 'Aircraft.Category',
 'Registration.Number',
 'Make',
 'Model',
 'Amateur.Built',
 'Engine.Type',
 'FAR.Description',
 'Schedule',
 'Purpose.of.flight',
 'Air.carrier',
 'Weather.Condition',
 'Broad.phase.of.flight',
 'Report.Status',
 'Publication.Date']

In [140]:
numerical_cols

['Number.of.Engines',
 'Total.Fatal.Injuries',
 'Total.Serious.Injuries',
 'Total.Minor.Injuries',
 'Total.Uninjured']

# Handling Missing Data

In [141]:
# Checking for missing data

df.isna().sum()

Event.Id                   1459
Investigation.Type            0
Accident.Number            1459
Event.Date                 1459
Location                   1511
Country                    1685
Latitude                  55966
Longitude                 55975
Airport.Code              40099
Airport.Name              37558
Injury.Severity            2459
Aircraft.damage            4653
Aircraft.Category         58061
Registration.Number        2776
Make                       1522
Model                      1551
Amateur.Built              1561
Number.of.Engines          7543
Engine.Type                8536
FAR.Description           58325
Schedule                  77766
Purpose.of.flight          7651
Air.carrier               73700
Total.Fatal.Injuries      12860
Total.Serious.Injuries    13969
Total.Minor.Injuries      13392
Total.Uninjured            7371
Weather.Condition          5951
Broad.phase.of.flight     28624
Report.Status              7840
Publication.Date          16689
dtype: i

In [142]:
# replace placeholders with actual NaN
df.replace(['N/A', 'NA', 'Na', 'NONE', 'Unknown', '--', ''], pd.NA, inplace=True)

In [143]:
# dropping rows missing critical identifiers fields
df.dropna(subset=['Make','Model'], inplace=True)

In [144]:
# Replacing missing categorical data with 'unknown'
for col in categorical_cols:
    df[col].fillna('Unknown', inplace=True)

In [145]:
# fill missing no.of.engines by grouping the make and model this preserves manufacturers level patterns
df['Number.of.Engines'] = df.groupby(['Make','Model'])['Number.of.Engines'].transform(lambda x: x.fillna(x.median()))
df['Number.of.Engines'].fillna(df['Number.of.Engines'].median(), inplace=True)
df['Number.of.Engines'].isna().sum()

0

In [146]:
injuries = ['Total.Fatal.Injuries', 'Total.Serious.Injuries', 
           'Total.Minor.Injuries', 'Total.Uninjured']
for col in injuries:
    median_val = df[col].median()
    df[col].fillna(median_val, inplace=True)



In [147]:
#Standardizing Categorical Data
exclude = ['Event.Id', 'Accident.Number','Registration.Number', 'Model' ]
for col in [c for c in categorical_cols if c not in exclude]:
    df[col] = (
        df[col]
        .astype(str)                
        .str.strip()                
        .str.title()                
    )



In [148]:
df.head()

Event.Id Investigation.Type Accident.Number Event.Date  \
0  20001218X45444           Accident      SEA87LA080 1948-10-24   
1  20001218X45447           Accident      LAX94LA336 1962-07-19   
2  20061025X01555           Accident      NYC07LA005 1974-08-30   
3  20001218X45448           Accident      LAX96LA321 1977-06-19   
4  20041105X01764           Accident      CHI79FA064 1979-08-02   

          Location        Country   Latitude   Longitude Airport.Code  \
0  Moose Creek, Id  United States    Unknown     Unknown      Unknown   
1   Bridgeport, Ca  United States    Unknown     Unknown      Unknown   
2    Saltville, Va  United States  36.922223  -81.878056      Unknown   
3       Eureka, Ca  United States    Unknown     Unknown      Unknown   
4       Canton, Oh  United States    Unknown     Unknown      Unknown   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0      Unknown  ...          Personal     Unknown                  2.0   
1      Unknown  ...          Personal     Unknown                  4.0   
2      Unknown  ...          Personal     Unknown                  3.0   
3      Unknown  ...          Personal     Unknown                  2.0   
4      Unknown  ...          Personal     Unknown                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    0.0                  0.0             1.0   
3                    0.0                  0.0             0.0   
4                    2.0                  0.0             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               Unk                 Cruise  Probable Cause          Unknown  
1               Unk                Unknown  Probable Cause       19-09-1996  
2               Imc                 Cruise  Probable Cause       26-02-2007  
3               Imc                 Cruise  Probable Cause       12-09-2000  
4               Vmc               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [149]:
# Saving the cleaned dataset

df.to_csv("Clean_Aviation_Data.csv", index=False)

In [150]:
df.isna().sum()

Event.Id                  0
Investigation.Type        0
Accident.Number           0
Event.Date                0
Location                  0
Country                   0
Latitude                  0
Longitude                 0
Airport.Code              0
Airport.Name              0
Injury.Severity           0
Aircraft.damage           0
Aircraft.Category         0
Registration.Number       0
Make                      0
Model                     0
Amateur.Built             0
Number.of.Engines         0
Engine.Type               0
FAR.Description           0
Schedule                  0
Purpose.of.flight         0
Air.carrier               0
Total.Fatal.Injuries      0
Total.Serious.Injuries    0
Total.Minor.Injuries      0
Total.Uninjured           0
Weather.Condition         0
Broad.phase.of.flight     0
Report.Status             0
Publication.Date          0
dtype: int64

In [151]:
df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

# Data Analysis

Objectives:
Your company wants to start operating aircraft for commercial and private use but lacks aviation experience.
We’ll analyze accident data to identify:

1. Which aircraft are the safest (lowest risk)?

2. What factors increase or decrease accident severity?

3. What operational strategies reduce exposure to high-risk conditions?

# Which Aircraft Are Lowest Risk?

In [ ]:
# Aircraft Risk Analysis: Make + Model

import pandas as pd
import numpy as np

# Clean and prepare base columns
df['Make'] = df['Make'].astype(str).str.strip().str.title()
df['Model'] = df['Model'].astype(str).str.strip().str.upper()
df['Aircraft.damage'] = df['Aircraft.damage'].astype(str).str.strip().str.title()


# Create helper analysis columns


# (a) Severity Index (Weighted injury score)
df['Severity_Index'] = (
    df['Total.Fatal.Injuries'].fillna(0)*3 +
    df['Total.Serious.Injuries'].fillna(0)*2 +
    df['Total.Minor.Injuries'].fillna(0)*1
)

# (b) Any Fatal? 
df['Any_Fatal'] = np.where(df['Total.Fatal.Injuries'] > 0, 1, 0)

# (c) Map Aircraft Damage → numeric scale
damage_map = {
    'Destroyed': 3,
    'Substantial': 2,
    'Minor': 1,
    'None': 0
}
df['Damage_Score'] = df['Aircraft.damage'].map(damage_map).fillna(1)  

# 3. Aggregate risk metrics by Make + Model

model_risk = (
    df.groupby(['Make', 'Model'], dropna=False)
      .agg(
          Total_Accidents=('Event.Id', 'count'),
          Mean_Severity=('Severity_Index', 'mean'),
          Fatal_Rate=('Any_Fatal', 'mean'),
          Mean_Damage=('Damage_Score', 'mean')
      )
      .reset_index()
)

# Filter for meaningful sample size (>=15 accidents)
model_risk = model_risk[model_risk['Total_Accidents'] >= 15].copy()


# 4. Normalize metrics and compute Composite Risk

def minmax(series):
    smin, smax = series.min(), series.max()
    if smax == smin:
        return pd.Series(0, index=series.index)
    return (series - smin) / (smax - smin)

for c in ['Mean_Severity', 'Fatal_Rate', 'Mean_Damage']:
    model_risk[f'Norm_{c}'] = minmax(model_risk[c])

# Average normalized values → Composite Risk Score
model_risk['Composite_Risk'] = model_risk[
    ['Norm_Mean_Severity', 'Norm_Fatal_Rate', 'Norm_Mean_Damage']
].mean(axis=1)

# 5. Final presentation fields for Tableau
model_risk['Fatal_Rate_Pct']   = (model_risk['Fatal_Rate'] * 100).round(2)
model_risk['Composite_Risk_r'] = model_risk['Composite_Risk'].round(3)
model_risk['Make_Model']       = model_risk['Make'] + " " + model_risk['Model']

# Top 10 safest aircraft
lowest_risk = (
    model_risk.sort_values('Composite_Risk', ascending=True)
              .head(10)
              .loc[:, ['Make', 'Model', 'Total_Accidents',
                       'Mean_Severity', 'Fatal_Rate_Pct',
                       'Mean_Damage', 'Composite_Risk_r']]
)

print("\nTop 10 Safest Aircraft Models:")
print(lowest_risk.to_string(index=False))

# 6. Export clean CSVs for Tableau

model_risk.to_csv('aircraft_summary.csv', index=False)
lowest_risk.to_csv('aircraft_summary_top10.csv', index=False)




Top 10 Safest Aircraft Models:
          Make       Model  Total_Accidents  Mean_Severity  Fatal_Rate_Pct  Mean_Damage  Composite_Risk_r
        Airbus        A319               16       1.125000            0.00     1.000000             0.006
        Boeing         787               26       0.500000            0.00     1.076923             0.018
        Boeing     737-7H4               19       0.894737            0.00     1.105263             0.026
        Boeing         777               95       0.810526            0.00     1.126316             0.029
       Embraer   EMB-145LR               21       1.380952            0.00     1.190476             0.045
        Boeing     757-232               24       2.625000            0.00     1.208333             0.056
Bombardier Inc CL-600-2B19               18       0.388889            0.00     1.277778             0.057
        Boeing     717-200               20       1.450000            0.00     1.250000             0.058
        Boeing

# Findings
Grouping accident data by Make + Model reveals that composite risk (combined average of severity, fatality rate, and damage) is lowest for newer, large-transport jets such as the Boeing 747, 787, 737-7H4 and the Bombardier CL-600-2B19 regional jet. while smaller or older models (e.g., McDonnell Douglas DC-9, MD-80) show higher severity averages.

# Recommendation 1
Begin operations with modern transport-category aircraft (e.g., Boeing 737-800 or 787) for initial fleet acquisition.
These models show the lowest historical accident severity, suggesting the best risk-to-profit ratio for a new operator.

# Weather Conditions and Accident Severity

In [124]:
# Weather Analysis
weather_summary = (
    df.groupby('Weather.Condition')
      .agg(
          Total_Accidents=('Event.Id', 'count'),
          Mean_Severity=('Severity_Index', 'mean'),
          Fatal_Rate=('Any_Fatal', 'mean'),
          Mean_Damage=('Damage_Score', 'mean')
      )
      .reset_index()
      .sort_values('Mean_Severity', ascending=False)
)

weather_summary


Weather.Condition  Total_Accidents  Mean_Severity  Fatal_Rate  Mean_Damage
1               Unk             1116       8.522401    0.509857     2.392473
2           Unknown             4439       7.673350    0.339716     2.097319
0               Imc             5971       7.113046    0.579132     2.531067
3               Vmc            77224       1.740171    0.158409     2.152220

In [125]:
weather_summary.to_csv("factor_Weather_Condition.csv", index=False)


# Findings
Accidents occurring in IMC Weather conditions are roughly four times more severe and three-to-four times more likely to be fatal than those in VMC Weather conditions. Mean Damage also increases, indicating that aircraft in IMC crashes suffer more destruction.

# Recommendation 2
Invest early in IMC-capable aircraft, IFR-rated pilots, and weather-driven dispatch policies.

# How the phase of flight and purpose of flight relate to accident severity and fatality rates.

In [126]:
# Accident severity by phase of flight
phase_summary = (
    df.groupby('Broad.phase.of.flight')
      .agg(
          Total_Accidents=('Event.Id', 'count'),
          Mean_Severity=('Severity_Index', 'mean'),
          Fatal_Rate=('Any_Fatal', 'mean'),
          Mean_Damage=('Damage_Score', 'mean')
      )
      .reset_index()
      .sort_values('Mean_Severity', ascending=False)
)

phase_summary


Broad.phase.of.flight  Total_Accidents  Mean_Severity  Fatal_Rate  \
11               Unknown            27638       3.766734    0.237716   
1                  Climb             2030       3.668966    0.303941   
0               Approach             6533       2.733048    0.243380   
6            Maneuvering             8133       2.673429    0.390754   
2                 Cruise            10259       2.670533    0.271371   
7                  Other              119       2.537815    0.378151   
3                Descent             1885       2.483289    0.242440   
4              Go-Around             1353       2.334812    0.197339   
9                Takeoff            12475       1.933467    0.145892   
8               Standing              945       1.441270    0.113228   
10                  Taxi             1958       0.525536    0.020940   
5                Landing            15422       0.468616    0.018934   

    Mean_Damage  
11     2.109487  
1      2.319212  
0      2.263126  
6      2.443748  
2      2.308120  
7      2.218487  
3      2.253581  
4      2.282336  
9      2.206493  
8      1.949206  
10     1.903473  
5      2.026456

In [129]:
# Accident Severity by Purpose of flight
purpose_summary = (
    df.groupby('Purpose.of.flight')
      .agg(
          Total_Accidents=('Event.Id', 'count'),
          Mean_Severity=('Severity_Index', 'mean'),
          Fatal_Rate=('Any_Fatal', 'mean'),
          Mean_Damage=('Damage_Score', 'mean')
      )
      .reset_index()
      .sort_values('Mean_Severity', ascending=False)
)

purpose_summary['Purpose.of.flight'] = purpose_summary['Purpose.of.flight'].astype(str).str.strip()
purpose_summary['Fatal_Rate_Pct']    = (purpose_summary['Fatal_Rate']*100).round(1)
purpose_summary['Mean_Severity_r']   = purpose_summary['Mean_Severity'].round(2)
purpose_summary['Mean_Damage_r']     = purpose_summary['Mean_Damage'].round(2)


In [130]:
phase_summary.to_csv("factor_Broad_phase_of_flight.csv", index=False)
purpose_summary.to_csv("factor_Purpose_of_flight.csv", index=False)


# Findings
Grouped results show that severity and fatality are highest in Maneuvering, Climb, and Cruise phases, and lowest during Landing.
By purpose, Instructional and Aerial Application flights are least severe, while Personal and Business flights show higher severity and fatal rates. Accidents during high-energy or unstructured operations are far more severe.
Structured, supervised missions reduce the risk of catastrophic outcomes.

# Recommendations 3
Implement strict Standard Operating Procedures (SOPs), limit early operations to structured mission types (charter, training), and emphasize recurrent training.
This reduces operational risk, increases reliability, and strengthens regulatory compliance for a new aviation division.